### ISSUE TITLE
IDE fails to detect compiler errror E0277

### ISSUE THREAD

**Issue by afetisov** [intellij-rust/intellij-rust#6954]
## Environment

* **IntelliJ Rust plugin version:** 0.3.142.3705-203
* **Rust toolchain version:** 1.52.0-nightly (45b3c2851 2021-03-04) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2020.3.2 (CL-203.7148.70)
* **Operating system:** Linux 5.4.0-66-generic
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description

According to [the tracking issue](https://github.com/intellij-rust/intellij-rust/issues/886), Intellij-Rust should be able to detect [E0277](https://doc.rust-lang.org/error-index.html#E0277). This isn't true. It doesn't detect errors in the sample provided in the docs, and it doesn't detect it in thee example below.

## Steps to reproduce

```rust
struct Foo;
fn f() where Foo: Copy {}
```

Expected error: "The trait bound `Foo: std::marker::Copy` is not satisfied [E0277]. The trait `std::marker::Copy` is not implemented for `Foo`"

But no error is marked.

**Comment by afetisov** [comment]
Fixed in the current build.
